python/parquet_tools/common/errhandlers.py

```diff
@@ -0,0 +1,190 @@
+
+    # *************************************************************************
+    # *                                 s a y                                 *
+    # *************************************************************************
+    @classmethod
+    def say(cls, *args, **kwargs):
+        """Print a message to stderr.
+
+        Parameters
+        ----------
+        args : 'varies'
+            The are the positional argumentgs to the print function.
+        kwargs : 'varies'
+            The are the positional argumentgs to the print function.
+        """
+
+        # Simply print the message prefixed by out command name.
+        #
+        print(cls.cmdname, *args, file=sys.stderr, **kwargs)
+
+    # *************************************************************************
+    # *                           s e t _ d e b u g                           *
+    # *************************************************************************
+    @classmethod
+    def set_debug(cls, val):
+        """Set the debug parameter for exceptions.
+
+        Parameters
+        ----------
+        val : 'bool'
+            When True, a traceback is printed when a fata exception occurs.
+        """
+
+        cls.doDebug = val
+
+
+# *****************************************************************************
+# *                            F a t a l E r r o r                            *
+# *****************************************************************************
+
+class FatalError(Exception, ErrInfo):
+    """Handle fatal exception errors.
+
+    Parameters
+    ----------
+    ex : 'exception object'
+        The exception object that caused the fatal error.
+    args : 'print style arguments"
+        Printable tokens providing context for the error.
+    """
+
+    def __init__(self, ex, *args):
+
+        # Produce a stack trace if debuggging is enabled.
+        #
+        if ErrInfo.doDebug:
+            traceback.print_exception(type(ex), ex, ex.__traceback__)
+
+        # Construct an informative message.
+        #
+        msg = ' '.join(str(x) for x in args) + '; fatal ' +\
+              type(ex).__name__ + ' exception: ' + str(ex)
+
+        # Initialize the exception object with the message.
+        #
+        Exception.__init__(self, msg)
+
+    def __context__(self):
+        pass
+
+
+# *****************************************************************************
+# *                             P a r m E r r o r                             *
+# *****************************************************************************
+
+class ParmError(Exception, ErrInfo):
+    """Handle fatal processing errors.
+
+    Parameters
+    ----------
+    msgnum : 'int'
+        The numeric message code to format an error message using args.
+    args : 'print style arguments"
+        Printable tokens providing context for the error.
+
+    Notes
+    _____
+    - The following message format numbers are defined:
```

Comment:
why the leading dash?

Reply:
I saw that in an example. The dash was being used as a bullet point.